Sources/MapboxNavigation/NavigationMapView.swift

```diff
@@ -392,9 +406,11 @@ open class NavigationMapView: UIView {
         if let legIndex = currentLegIndex {
             let congestionFeatures = route.congestionFeatures(legIndex: legIndex,
                                                               roadClassesWithOverriddenCongestionLevels: roadClassesWithOverriddenCongestionLevels)
-            currentLineGradientStops = routeLineGradient(congestionFeatures,
-                                                         fractionTraveled: fractionTraveled,
-                                                         isSoft: crossfadesCongestionSegments)
+            currentLineGradientStops = routeLineCongestionGradient(congestionFeatures,
+                                                                   fractionTraveled: fractionTraveled,
+                                                                   isSoft: crossfadesCongestionSegments)
+            currentRestrictedAreasStops = routeLineRestrictionStops(route.restrictedRoadsFeatures(),
```

Comment:
Because we only need the `currentRestrictedAreasStops` when `showsRestrictedAreasOnRoute` enabled, but the `navigationMapView.setUpLineGradientStops(along:)` is called when `routeLineTracksTraversal` enabled. It would be better if we could move it under the check of `showsRestrictedAreasOnRoute` in `navigationMapView.show(_:legIndex:)` and the setter of `showsRestrictedAreasOnRoute`.

Reply:
Good catch! Corrected